Bug 1571411 - Initial test validity dashboard

airimovici opened the pull request:
[Bugzilla bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1571411)

octavian-negru commented on ui/perfherder/health/HealthTableControls.jsx:
```diff
@@ -0,0 +1,96 @@
+
+  updateFilteredResults = () => {
+    const { filterText } = this.state;
+
```
Remove this space please.

octavian-negru commented:
I have some opinions related to page structure / behaviour:
1. It would be nice to have two navigation bars. The one below the table and one above. This is just an opinion.
2. Input filtering must filter from suite also? Now it's filtering only by test name.
3. It would be a great effort to add a new query param with the filter from input? On the Perfherder's Compare page, putting parameters in URL is a convenient way to share informations.
4. Empty rows 
    1. Select a talos (just an example), must have < 100 rows
    2. From navbar select 100 rows
    
    Empty rows are shown. If possible, make the React Table to show only the rows with data.

octavian-negru commented on treeherder/webapp/api/machine_platforms.py:
```diff
@@ -0,0 +1,16 @@
+        A ViewSet for listing all the machine platforms.
+    """
+    def list(self, request):
+        queryset = MachinePlatform.objects.all()
```
I think `queryset` and `serializer` are a bit too generic.

octavian-negru commented on ui/perfherder/health/HealthView.jsx:
```diff
@@ -0,0 +1,151 @@
+  }
+
+  componentDidMount() {
+    const { projects, platforms, updateAppState } = this.props;
```
I think the code from componentDidMount related to `platforms` can be moved into a different function. It would make the coude decoupled. The function would handle, separately, `platforms` creation. Also, it would be easy testable.

airimovici commented on ui/perfherder/health/HealthView.jsx:
```diff
@@ -0,0 +1,151 @@
+      });
+    }
+    // create projectsMap
+    this.createObjectsMap(projects, 'projectsMap', 'name');
```
I'm not able to repro this

ionutgoldan replied:
Sorry, I've been using `yarn start`...

airimovici commented on ui/perfherder/health/HealthTable.jsx:
```diff
@@ -0,0 +1,105 @@
+            const platforms = props.value.map(id => platformsMap[id]);
+            return <BadgeList items={platforms} color="info" />;
+          }
+          return <Spinner color="info" />;
```
The cog spinner is not that configurable. I used the reactstrap default spinner :) we can revisit this if we decide it's not looking good

sarah-clements replied:
I'm not sure what you mean that it's not that configurable. You can use it in the <LoadingSpinner /> component and if there is a particular change you want to make, you can easily refactor it to pass in a prop. We should keep it consistent, as Ionut said.

airimovici replied:
By not configurable I mean that it needs some changes to use it as I wanted to use it (for showing that part of the data is loading), as now it's thought as a loader for the page. What I wanted to achieve is to show this spinner only when that data takes more to load. I understand that it is inconsistent with the rest of the application, so I will change it to fall under the page spinner :)

airimovici commented on treeherder/webapp/api/machine_platforms.py:
```diff
@@ -0,0 +1,16 @@
+from .serializers import MachinePlatformSerializer
+
+
+class MachinePlatformsViewSet(viewsets.ViewSet):
```
This can be done as a `good first bug` if you think it is suitable

camd replied:
``refdata`` was a term we used early on which stands for "Reference Data".  This was intended to be things related to jobs, like "machine" or "option collection" or "Job Type", etc.  

We used to use the [``ReferenceDataSignatures``](https://github.com/mozilla/treeherder/blob/7fd8bd2e00c9b0519b9a342dfceb330eeea76f9b/treeherder/model/models.py#L376) to allow sheriffs to hide certain jobs they wanted to ignore.  We don't do that anymore, however.  Now we just assign them to Tier-3, so they're hidden by default.  

We DO still create that signature to help users filter the list of jobs more specifically than by just the text strings (see the ``Job`` field in the left side of the details panel for a ``sig`` link.

There's lots more history I could give you, but, it's really moot at this point.  :)  The "Reference Data" term could be removed if it was cumbersome in any way.  It's just a general naming convention.

airimovici commented on ui/perfherder/App.jsx:
```diff
@@ -205,6 +208,30 @@ class App extends React.Component {
+                  />
+                )}
+              />
+              <Route
```
I'm not sure about this. I saw that for the other pages were created routes like this, maybe @sarah-clements can help us with some info here :)

sarah-clements replied:
It seems redundant but it is actually needed. Note the path is different - this is how react router handles use of query params.

airimovici commented on ui/perfherder/health/BadgeList.jsx:
```diff
@@ -0,0 +1,37 @@
+import PropTypes from 'prop-types';
+import { Badge } from 'reactstrap';
+
+export default function BadgeList(props) {
```
This can be a good first bug if you think it's suitable

airimovici commented:
> I have some opinions related to page structure / behaviour:
> 
> 1. It would be nice to have two navigation bars. The one below the table and one above. This is just an opinion.
> 2. Input filtering must filter from suite also? Now it's filtering only by test name.
> 3. It would be a great effort to add a new query param with the filter from input? On the Perfherder's Compare page, putting parameters in URL is a convenient way to share informations. Again, it is just an opinion.
> 4. Empty rows
>    
>    1. Select a talos (just an example), must have < 100 rows
>    2. From navbar select 100 rows
>    
>    Empty rows are shown. If possible, make the React Table to show only the rows with data.
> 
> Edit: If the implementation of the above points affects our OKRs, then we can go with what we have and open improvement tickets later.

For 3. and 4. it's not in our current scope, but can be done as improvements with separate bugs

octavian-negru commented:
@airimovici, totally agree. For the points 3 and 4 we can open improvements bugs.

sarah-clements commented on ui/perfherder/health/HealthView.jsx:
```diff
@@ -0,0 +1,151 @@
+class HeathView extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.validated = this.props.validated;
```
This is needed if you're using the`withValidation` higher order component to validate common query params. Probably wouldn't be worth using for just one query param.

sarah-clements replied:
I think you're right this line isn't needed here.

airimovici commented:
> * Please use colored red/green text for Total Alerts - not as a block of color with white text - and icons for people with red/green colorblindness.

Those numerical values are purely informative. They don't have any significance of good/bad. I'm actually inclined to remove the coloring entirely.

airimovici commented:
@sarah-clements I did some visual changes as you suggested. 

For the Platforms and Projects I removed the colored background, but I added an outline, as I could not see the difference between the items(platforms or projects) without an outline.

sarah-clements commented:
> @sarah-clements I did some visual changes as you suggested.
> 
> For the Platforms and Projects I removed the colored background, but I added an outline, as I could not see the difference between the items(platforms or projects) without an outline.

Thanks for making the changes. Regarding these colored text boxes though, can you make them distinct by using colored or bold text without the outline? Especially since you're only showing two for both Platforms and Projects (using an ellipsis for the remaining), I'm wondering why they'd need to command that much attention with the styling. :)

airimovici commented:
Hopefully I understood it correctly, this is the before and after:
Before:
![health-outline](https://user-images.githubusercontent.com/51707553/69557036-95318c80-0fae-11ea-809f-be57b9fdb7ff.png)

After:
![health-no-outline](https://user-images.githubusercontent.com/51707553/69557058-a2e71200-0fae-11ea-9e91-eb3dae9deb3b.png)

airimovici commented:
@sarah-clements I've done some visual changes and I posted some screenshots above

sarah-clements commented:
> @sarah-clements I've done some visual changes and I posted some screenshots above

That does look better but I think having them bold _and_ colored makes them look like links. What do you think about changing the font weight to normal and the font size to the same as the rest of the text in the table?

sarah-clements commented on ui/perfherder/health/HealthView.jsx:
```diff
@@ -0,0 +1,176 @@
+    const frameworkNames =
+      frameworks && frameworks.length ? frameworks.map(item => item.name) : [];
+
+    const healthDropdows = [
```
oops, forgot an 'n' :)

sarah-clements commented on ui/perfherder/health/HealthView.jsx:
```diff
@@ -0,0 +1,176 @@
+  location: PropTypes.shape({}),
+  validated: PropTypes.shape({
+    projects: PropTypes.arrayOf(PropTypes.shape({})),
+    frameworks: PropTypes.arrayOf(PropTypes.shape({})),
```
I think these two props are only passed through `withValidation` and are not actually part of the `validated` props.

sarah-clements commented on ui/perfherder/health/ItemBadge.jsx:
```diff
@@ -0,0 +1,25 @@
+    >
+      {itemText}
+    </span>
+  );
```
Will this component will be going away based on your recent screenshot? (FYI, for future use of badges there's a react strap [component](https://reactstrap.github.io/components/badge/) for that).

airimovici replied:
The first iteration used simple reactstrap badges (when it had colored background), but after that, I created the ItemBadge component to have the same feel of the simple badge, but with the outline(it does not come with outline in reactstrap). Now that I deleted the outline, I forgot to simplify the component (maybe remove it altogether). Thanks for pointing it out :+1:

airimovici commented on ui/css/perf.css:
```diff
@@ -408,6 +408,23 @@ button:disabled {
+.item-outlined {
+  border: 1px solid;
+  border-radius: 5px;
+}
```
No, I deleted the class

sarah-clements commented on ui/css/perf.css:
```diff
@@ -409,20 +409,8 @@ button:disabled {
-  font-size: 12px;
-  font-weight: 600;
   padding: 3px 6px;
   color: #17a2b8;
```
I think you can replace this`item-badge` class with the bootstrap color and padding utilities (`text-info`).

sarah-clements commented on ui/perfherder/health/HealthTable.jsx:
```diff
@@ -0,0 +1,98 @@
+      Cell: props => {
+        if (platformsMap) {
+          const platforms = props.value.map(id => platformsMap[id]);
+          return <ItemList items={platforms} color="info" />;
```
I saw a console warning that this `ItemList` is missing a key.

airimovici commented:
Thanks for the suggestions :) I didn't have a chance to resolve the last 2 before this got merged. I'll create a small PR after this one gets deployed
